lib/reline/unicode.rb

```diff
@@ -160,24 +160,29 @@ def self.split_by_width(str, max_width, encoding = str.encoding)
     width = 0
     rest = str.encode(Encoding::UTF_8)
     in_zero_width = false
+    seq = String.new(encoding: encoding)
     rest.scan(WIDTH_SCANNER) do |gc|
```

Comment:
Does the gc here mean "graphic cluster"?

Reply:
I think it was grapheme cluster.

https://github.com/aycabta/reline/blob/f348ecd9f507db62f3ac93619f3aef40b46993f6/lib/reline/line_editor.rb#L283
```ruby
@line.encode(Encoding::UTF_8).grapheme_clusters.each do |gc|
```

Reply:
The name `gc` was given in this commit.
https://github.com/ruby/reline/commit/29175733633afaea9aff5d467598338fda6a56d3

But grapheme clusters are Unicode context, so it is different from the terminal escape sequences (NON_PRINTING_START, NON_PRINTING_END, CSI_REGEXP_INDEX, and OSC_REGEXP_INDEX). It is a little complicated.
`gc` mean grapheme clusters, but I think actual `gc` mean about a string divided by WIDTH_SCANNER.

Reply:
Thanks for both of your explanation!

Yeah I think calling the array `gc` is a big misleading. After this is merged, I want to open a PR to expand the `gc` (array) into:

```rb
@line.encode(Encoding::UTF_8).grapheme_clusters.each do |non_printing_start, non_printing_end, csi, osc, gc|
```

Then we can get rid off of the index constants and a few re-assignments.